### ISSUE TITLE
Problem in Adding Response Headers --- Server Stops Sending Packets

### ISSUE THREAD

**Issue by FatimaZar** [tatsuhiro-t/nghttp2#465]
Hi,

I am using the nghttpd server (version-1.6.0) for my work and I would like to:

(1)  disable the cache-control response header and 
(2) add the response header Access-Control-Allow-Origin: *  and another header of this form.

I modified the HttpServer.cc file, and in particular the "Http2Handler::submit_file_response" method, and I compiled again using the same approach described in the README.rst. However, the problem is that **adding any new response header stops the server from sending any packets to the client.**

When I removed the cache-control response header the server still worked. However, adding new headers is causing the server to close the stream. Please note that the same change that I made here works in nghttp2's version 1.0.5. But, I don't know why my changes do not work with this version. 

I will highly appreciate any help to fix this problem. Thank you !

Please find my code below:

 auto nva = make_array(http2::make_nv_ls(":status", status),
                        http2::make_nv_ll("server", NGHTTPD_SERVER),
                        http2::make_nv_ls("content-length", content_length),
			http2::make_nv_ls("date", sessions_->get_cached_date()),
			http2::make_nv_ll("Access-Control-Allow-Origin", "*"), 
                        http2::make_nv_ll("", ""), 
			http2::make_nv_ll("", ""), http2::make_nv_ll("", ""));

Best regards,
Fatima

**Comment by tatsuhiro-t** [comment]
Did you update `size_t nvlen = 5;` when the initial number of headers are changed?
Since this issue is caused by your change, I have no idea what is going on.
Could you give us complete diff what you made a change?

**Comment by FatimaZar** [comment]
Hi Mr. Tatsuhiro,

Thank you for your quick response. 

Actually,  I do have size_t nvlen = 5; . 

The statement in my first post (assignment to nva) is the only thing that I changed. All I did was remove the line "http2::make_nv_ll("cache-control", "max-age=3600")," and add the line   "http2::make_nv_ll("Access-Control-Allow-Origin", "*"),"    

Since, the nvlen did not change, I kept it 5. 

Also, when compiling, I am using:

./configure --enable-asio-lib --disable-examples
and then make

The above are the only things that I am doing differently. I have not changed anything else in the code.

Thank you,
Fatima
